### ISSUE TITLE
`code` and `message` is the same in JSON output 

### ISSUE THREAD

**Issue by potch** [mozilla/addons-linter#1287]
### Describe the problem and steps to reproduce it:
Get results from programmatic usage of the linter (as an API)

### What happened?
In results, the `code` field of warnings is the same thing as the `message` field. example:
```js
{
  _type: 'warning',
  code: 'The Function constructor is eval.',
  message: 'The Function constructor is eval.',
  ...
}
```

### What did you expect to happen?
The `code` field that would be something like "UNSUPPORTED_API", as listed in the messages definition: https://github.com/mozilla/addons-linter/blob/286ebbc91025baa19fcb8eace30d28144193f966/src/messages/javascript.js#L266

### Anything else we should know?
Nah.

**Comment by mstriemer** [comment]
Since `code` is something we added and isn't supported when adding an error through eslint I think we're stuck with this for now.

@EnTeQuAk and I discussed moving to the eslint standard of using the rule name to replace `code` and we agreed that was the best path forward but I don't see an issue for that.

**Comment by EnTeQuAk** [comment]
Correct, it's a bit of a tricky problem. We do have `code` and `message` and at the moment use both and for our internal messages `code` is set properly but for messages coming from eslint it's set to `message` which isn't a problem at the moment since more or less `code` was unused.

We do have the ability to overwrite `code` for messages coming from eslint, we should probably do and require that. I do agree that having a `code` is useful longerm since it allows ignoring or filtering messages on the receiving client (AMO, web-ext etc).

**Comment by potch** [comment]
I'm looking to filter for specific codes to filter under programmatic usage. Can I pass a flag that's like "this isn't eslint, give me generalized codes"?

**Comment by EnTeQuAk** [comment]
I marked this as fixed with merging https://github.com/mozilla/addons-linter/pull/1356. We are now able to overwrite the `code` for a 3rd party rule as well, we are currently overwriting all 3rd party rules so as far as I can tell this issue should be fixed now.

Mind testing if this helps with what you wanted to do @potch?

**Comment by vcarciu** [comment]
Please add some STRs to this bug or mark it as "qa: not needed"
